Archive/Imo/Imo1959Q2.lean

```diff
@@ -39,7 +40,7 @@ theorem isGood_iff : IsGood x A ↔
     sqrt (2 * x - 1) + 1 + |sqrt (2 * x - 1) - 1| = A * sqrt 2 ∧ 1 / 2 ≤ x := by
   cases le_or_lt (1 / 2) x with
   | inl hx =>
-    have hx' : 0 ≤ 2 * x - 1 := by linarith
+    have hx' : 0 ≤ 2 * x - 1 := by linear_combination 2 * hx
```

Comment:
In terms of readability, these changes feel like a small net negative to me. Speed improvements could make it worth again.